api/v1beta2/ibmvpccluster_types.go

```diff
@@ -32,28 +32,39 @@ const (
 
 // IBMVPCClusterSpec defines the desired state of IBMVPCCluster.
 type IBMVPCClusterSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// The IBM Cloud Region the cluster lives in.
+	// controlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint capiv1beta1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// ControlPlaneLoadBalancer is optional configuration for customizing control plane behavior.
+	// +optional
+	// dep: use LoadBalancers instead
+	ControlPlaneLoadBalancer *VPCLoadBalancerSpec `json:"controlPlaneLoadBalancer,omitempty"`
+
+	// cosInstance is the IBM COS instance to use for cluster resources.
+	COSInstance *COSInstance `json:"cosInstance,omitempty"`
```

Comment:
If the shared basic types is added in time, this should be `COSInstanceReference`, from `types.go`.

Reply:
I'm dropping this reference, as I don't plan to manage COS resources.

Instead, I will incorporate a new `ImageSpec`, which will have details about the expected VSI Custom Image, which could include a COS ObjectURL, to create a new Custom Image from, if necessary.